addon/index.ts

```diff
@@ -156,28 +154,26 @@ function applyOptions(
   options: TaskGroupOptions | TaskOptions,
   task: TaskGroupProperty | TaskProperty
 ): (TaskGroupProperty | TaskProperty) & Decorator {
-  return Object.entries(options).reduce(
-    (
-      taskProperty,
-      [key, value]: [
-        keyof typeof options,
-        ObjectValues<Required<typeof options>>
-      ]
-    ) => {
-      assert(
-        `ember-concurrency-decorators: Option '${key}' is not a valid function`,
-        typeof taskProperty[key] === 'function'
-      );
-      if (value === true) {
-        return (taskProperty[key] as () => typeof taskProperty)();
-      }
-      return (taskProperty[key] as (o: typeof value) => typeof taskProperty)(
-        value
-      );
-    },
-    task
-    // The CP decorator gets executed in `createDecorator`
-  ) as typeof task & Decorator;
+  const keys = Object.keys(options) as Array<keyof typeof options>;
+
+  for (const key of keys) {
+    const value = options[key];
```

Comment:
should this be something like the below snippet to keep the semantics the same?

```suggestion
    const value: ObjectValues<Required<typeof options>> = options[key];
```